### ISSUE TITLE
First setup (on mac)

### ISSUE THREAD

**Issue by ringvold** [PhlexPlexico/G5API#33]
Hi! 

I'm trying to set up the application locally to test it out and have it ready for the frontend that I plan to create for it. I have tried the old webpanel for get5 (original and your version) so I sort of know how that works but not looked far into how it worked. I'm a fairly experienced developer, but has not worked much with node as a backend.

This is my experience and thoughts on the process so take it as you wish.  😀

I'm trying to run the application locally on my mac. I have other machines with Linux and Windows (WSL) which I will try out later so I'll report on that experience if that is of interest.

## Setup instructions

I have tried and failed, read the readme a bit more thoroughly and then got some things. 😅 Even if the formation was in the readme I end but not finding it the first time thorugh so think it could be structured a bit more clearly. Maybe more of a step by step guide for the setup?
I have thoughts on this that could be a separate issue if you are interested.

On mac I used homebrew to install mariadb and redis:
`brew install mariadb redis`

Redis needed some configuration as it seems to default to NOAUTH which this app does not accept.
I added `requirepass super_secure` ("super_secure" is the chosen password in this example) in the redis config file (/usr/local/etc/redis.conf) to make it work.

## Enviroment and database config

At first glance the config file and envrionment setup seems fine, but when it came to the database setup some things tripped me up. There is a section for the database. That is fine (whats with the capicalization on `Database` and `Server` though?), but  there is also the environment section (`dev` in development.json, `test` in testing etc). It seems to be used by the db migrations scripts.

I made the migrations run fine after some tweaks to the sql user creation commands in the readme but when I started the server I got db auth problems. This seems to be because, as mentioned above, the migrations use one part of the config file and the app another part so the username and password need to be duplicated. This was not very clear until I hit this particular wall and poked around. I first thought that the `Server` part was not in use and only the `dev` part of the config was in use.

It seems to be because db-migrate has another environment config than you are using for the project. Maybe both could default to get the config from environment variables? Or maybe there is another way to make the to environment setups work together so the setup is a bit more intuitive?

## Final thoughts

I think it would be nice to have a docker setup for the application. It would make it easier to get started and is also a way to have one way to set up and run the application. Problems are easier to reproduce and people does not have to install databases locally and deal with differences in default configurations (ref. redis AUTH).

With all this said I got to give it to you for a well structured and clean application. It seems easy to navigate and the code is well documented! 😄 👏

**Comment by PhlexPlexico** [comment]
Hey! Thanks for this info! I'll move this over to the wiki with some more step-by-step, it was kind of the last minute thing just in case some people wanted to take a peak at the application without delving into it so much, since I didn't expect anyone to use it at the time (when it was a proof-of-concept).  I thought about spinning out the config to one portion of the file, and you're right in the comment about it being db-migrate. There's a few things I haven't tested that I should now go back to since the bare-bone implementation  is complete, such as changing what the environment for `db-migrate` uses. 

I'm glad you were able to get it working relatively smoothly! This is actually **super** helpful in terms of setup and documentation, that I hope with these comments I can make some changes in the documentation to allow for a more smooth setup. I've never really delved into Docker files all that much, but I can also see what I can do, since there are plenty of examples around GitHub to take a look at!

Thank you very much for the input!

**Comment by PhlexPlexico** [comment]
Okay, latest push out here should update the configs to make it a bit less confusing.

I've also adjusted the "Server" section to "server", that was an old thing that I never really realized was much of a problem, just bad casing :)

Make sure to update your configs to reflect the changes! I've been doing testing largely on macOS and WSL (Ubuntu) and it's been running just fine here. However, since I've been working since scratch there is probably a few changes that I've made and forgotten to document. I'll start working on a wiki page for setup in the near future here, so be sure to keep an eye out on the Wiki! Thanks again! 😃
